### ISSUE TITLE
BUG:Good thing for moose2 is much more category and bad thing for moose 2 is low performance.

### ISSUE THREAD

**Issue by YixinChen-AI** [ENHANCE-PET/MOOSE#92]
I have followed your work for a while and Im a researcher in the similar filed. I just guess that if you use totalseg dataset to train the moose2 zoo. I just feel that the ribs and lung and some new categories suffer a low performance. 

**I am not sure if I send the wrong direction CT image. I seed the standard direction CT image as input (ct_itk.GetDirection()=(1,0,0,0,1,0,0,0,1) with 1.5 mm spacing)**. I use Moose2 to inference the low dose CT image, which is the data from 2016 low dose CT image challenge.

However, I just feel the performance is not as good as I expected since moose1 was trained in a supervised way.

In summary, I just want to mention that **if you use the totalseg dataset to construct moose2, totalseg was a full dose CT dataset and there might be a domain gap.**

Generally, MOOSE is a amazing tool. PS: i also has a question, if I want to compare with MOOSE in my research, should I compare with MOOSE1 or MOOSE2...

**Comment by LalithShiyam** [comment]
Hi @YixinChen-AI many thanks for bringing this up. Just to make things clear, we don't use total-segmentator's open access data, mainly because of reference segmentation accuracy. 

We use our own datasets, since we accumulated quite a bit of CTs and ALL of them are low dose CT, from our whole-body PET/CT studies. 

I can understand why ribs are suboptimal now, because we are hyper-segmenting ribs in `moosev2` whereas in `moosev1` it was the entire rib. This is much easier to segment. So errors can happen. The same can be mentioned about lungs as well. We are now providing 5x segments in `v2` and not the entire lung like in `v1`. If you can share one of the open datasets that you used, I can look into it in detail. Also we will be dropping a new model soon, with a bigger dataset. May be you find the accuracy acceptable there. 

Cheers,
Lalith

**Comment by YixinChen-AI** [comment]
Thanks you for your kind reply. That must be a huge annotation cost if you did it on the low dose CT. Thank you again for your effort for the low dose CT.

based on what you reply, the performance issue might not be the domain shift between CT and LDCT. I use moose2 model clin_ct_ribs and organs and other models, except PUMA and all_bone_v1 to inference the 10 training data on 2016 low dose CT challenge from mayo clinic (https://aapm.app.box.com/s/eaw4jddb53keg1bptavvvd1sf4x3pe9h). I just found the performance is not as expected on the 1mm_D30 quarter dose CT (50mAs). I have resample the data to 1.5mm and made sure that the direction is (1,0,0,0,1,0,0,0,1). However, the performance is not high. 

If you train all moose2 models  on your LDCT dataset, I think the performance of the models are good. However, I am not sure if the different equipments cause this since you might use whole-body (WB) data to train and that data (low performance) is not collected from WB machine. 

PS: i still have a concern that why you choose to train multiple models and not to train one big model, including organs, ribs and other regions.

Yours
Y Chen

**Comment by LalithShiyam** [comment]
Your question already had the answer. It's a huge undertaking and for logistical reasons we choose individual subsets rather than the entire tissue class. Also the memory burden increases if we go for a full stack of tissues and most of our users felt it wasn't needed. 

Our whole body dataset is a mixture of datasets from different PET/CT systems. The acquisition was just whole body. Also you don't need to resample it to 1.5 mm. `Moosev2` does it for you.

**Comment by Keyn34** [comment]
@LalithShiyam sure, I will take a look at that!

@YixinChen-AI, I will use the data you linked as raw input for MOOSE and will let you know if I notice something is off as well.

**Comment by YixinChen-AI** [comment]
Thank you guys!

**Comment by LalithShiyam** [comment]
Closing this due to inactivity.
